### ISSUE TITLE
glorotUniform's depth scaling not actually used in transformer's FFN 

### ISSUE THREAD

**Issue by fiqas** [marian-nmt/marian-dev#861]
### Bug description
In transformer.h, I noticed that now we're scaling glorotUniform by depth or something.

`auto initFn = inits::glorotUniform(true, true, depthScaling_ ? 1.f / sqrtf((float)depth_) : 1.f);`

It's passed into `denseInline` function, but is not used in the end, just a normal glorot:

`auto W = graph->param(prefix + "_W" + suffix, {x->shape()[-1], outDim}, inits::glorotUniform()); `

 I'd say it's a bug as if one wants to use that depth scaling, it won't actually be used in FFN after all.

I think it should be replaced with:

`auto W = graph->param(prefix + "_W" + suffix, {x->shape()[-1], outDim}, initFn);`

but I'm unsure of that feature implementation. It is used for attention, so I guess it should be used in FFN too?

### Context
```
../build_reg/marian --version
v1.10.18; a0ef8e6 2021-04-19 17:14:05 +0000
```

**Comment by emjotde** [comment]
Hi,
where exactly, file, line number?

**Comment by emjotde** [comment]
Oh, got it. Good catch. How is that even compiling? Should die with a warning about unused parameters.

**Comment by emjotde** [comment]
Fixed in 3e51ff38
Thanks again.
